app/models/manageiq/providers/autosde/storage_manager/cloud_volume.rb

```diff
@@ -38,13 +38,13 @@ def raw_update_volume(options = {})
       :size => options["size_GB"]
     )
     ext_management_system.autosde_client.VolumeApi.volumes_pk_put(ems_ref, update_details)
-    EmsRefresh.queue_refresh(ems)
+    cloud_vol = ext_management_system.cloud_volumes.find_by(:ems_ref=>ems_ref)
+    EmsRefresh.refresh(cloud_vol)
```

Comment:
You actually don't need to do another query to find the cloud-volume, since this is an instance method `self` already is the record that you're looking for.

Also you don't want to do `EmsRefresh.refresh` because that will run it in the current worker (probably a generic worker) rather than the refresh worker.  All refreshes have to be run by the refresh worker in order to ensure data consistency.

```suggestion
    EmsRefresh.queue_refresh(self)
```